lib/mayaUsd/resources/ae/usdschemabase/ae_template.py

```diff
@@ -660,6 +659,123 @@ def get(self):
             folderIndex[groups].items.append(attributeInfo.name)
         return self._attributeLayout
 
+class DisplayCustomDataControl(object):
+    '''Custom control for adding in extra display custom data.'''
+    def __init__(self, item, prim):
+        super(DisplayCustomDataControl, self).__init__()
+        self.item = item
+        self.prim = prim
+
+        # Check whether Ufe has the scene item meta data support.
+        self.useMetadata = hasattr(ufe.SceneItem, "getGroupMetadata") and hasattr(ufe.Value, "typeName")
+
+    @property
+    def GROUP(self):
+        # When using meta data we use a special group name which will automatically
+        # write to the session layer.
+        return 'SessionLayer-Autodesk' if self.useMetadata else 'Autodesk'
+
+    @property
+    def USE_OUTLINER_COLOR(self):
+        # To be more generic and not Maya specific we write out the custom data
+        # as "Text Color" instead of "Outliner Color".
+        key = 'Use Text Color'
+        return key if self.useMetadata else (self.GROUP + ':' + key)
+
+    @property
+    def OUTLINER_COLOR(self):
+        key = 'Text Color'
+        return key if self.useMetadata else (self.GROUP + ':' + key)
```

Comment:
When using metadata we call the group method, otherwise we compose the group key ourselves.
In Maya even though the fields are "Outliner Color" we write the custom data using "Text Color" to be more generic and sharable with Max.